tests/from_csv_test.py

```diff
@@ -0,0 +1,81 @@
+import glob
+import os
+
+import pytest
+import vaex
+
+path = os.path.dirname(__file__)
+csv_path = os.path.join(path, 'data', 'small3.csv')
+
+
+def test_from_csv():
+    # can read with default options
+    df = vaex.from_csv(csv_path)
+    _assert_csv_content(df, with_index=True)
+
+    # can read an empty CSV
+    df = vaex.from_csv(os.path.join(path, 'data', 'empty.csv'), copy_index=False)
+    assert len(df) == 0
+
+    # can read as chunks iterator
+    df_iterator = vaex.from_csv(csv_path, chunk_size=1, copy_index=False)
+    df1 = next(df_iterator)
+    assert len(df1) == 1
+    df2, df3 = next(df_iterator), next(df_iterator)
+    with pytest.raises(StopIteration):
+        next(df_iterator)
+    _assert_csv_content(vaex.dataframe.DataFrameConcatenated([df1, df2, df3]))
+
+
+def test_from_csv_converting_in_chunks():
+    _cleanup_generated_files()
+
+    # can read several chunks with converting, intermediate files are deleted
+    df = vaex.from_csv(csv_path, copy_index=False, chunk_size=1, convert=True)
+    _assert_csv_content(df)
+    for filename in ['small3.csv_chunk0.hdf5', 'small3.csv_chunk1.hdf5', 'small3.csv_chunk2.hdf5']:
+        assert not os.path.exists(os.path.join(path, 'data', filename))
+    assert os.path.exists(os.path.join(path, 'data', 'small3.csv.hdf5'))
+    _cleanup_generated_files(df)
```

Comment:
@maartenbreddels I continue having troubles trying to make this test work on Windows. First I was not closing DataFrames properly, but now I'm getting a segmentation fault from deeper DataFrame threading logic: see [this run](https://github.com/vaexio/vaex/pull/611/checks?check_run_id=478452044#). Do you have any ideas how to fix it? I've seen that in the `open_test.py` the Windows tests [are just skipped](https://github.com/vaexio/vaex/blob/master/tests/open_test.py#L12), should I do the same here?

Reply:
Difficult to say if we should skip it, basically I didn't care about windows before, but I care a bit more now, as we do have serious users. Do you have access to a windows virtual machine maybe?
I am also not sure if the segfault is related, I may have seen that before (which is scary), maybe some external library got updated.

Reply:
> Difficult to say if we should skip it, basically I didn't care about windows before, but I care a bit more now, as we do have serious users. Do you have access to a windows virtual machine maybe?
> I am also not sure if the segfault is related, I may have seen that before (which is scary), maybe some external library got updated.

it happened again, see [the build in appveyor](https://ci.appveyor.com/project/maartenbreddels/vaex-4gh4b/builds/31191907/job/648o04taq1w5hqre): with segfault in generic stacktrace, but seems to happen during/after the new `tests\from_csv_test.py`. The windows builds in GitHub Actions have succeeded. Do you have an idea how to proceed? Please advice, as I'm out of ideas :)